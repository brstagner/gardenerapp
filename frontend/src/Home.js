import { NavLink } from "react-router-dom";
import { Alert } from "reactstrap";

function Home ({ currUser }) {

    return (
        <div className="container">
            {currUser ?
                <h4>Welcome to the Gardener App.</h4> :
                <div className="container">
                    <div className="row gy-5">
                        <div className="col">
                            <div className="p-3">
                                <h4 className="d-flex justify-content-center">
                                    Have an account?
                                </h4>
                                <div className="d-flex justify-content-center">
                                    <NavLink className="btn btn-primary d-inline-block" to="/login">
                                        Login
                                    </NavLink>
                                </div>
                            </div>
                            <div className="p-3"></div>
                            <div className="p-3">
                                <h4 className="d-flex justify-content-center">
                                    New user?
                                </h4>
                                <div className="d-flex justify-content-center">
                                    <NavLink className="btn btn-primary d-inline-block" to="/register">
                                        Register
                                    </NavLink>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            }
        </div>
    );
}

export default Home;