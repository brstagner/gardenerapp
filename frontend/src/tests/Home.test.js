import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Home';

const user = { username: "user", userId: "1", isAdmin: "false", token: "user-token" };
const admin = { username: "admin", userId: "2", isAdmin: "true", token: "admin-token" };

// smoke test
it("renders without crashing", function () {
    render(
        <BrowserRouter >
            <Home />
        </BrowserRouter >);
    // screen.debug()
});

// snapshot test
it("matches snapshot for no logged-in user", function () {
    const { asFragment } = render(
        <BrowserRouter >
            <Home />
        </BrowserRouter >);
    expect(screen.getByText("New user?")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

// snapshot test
it("matches snapshot for logged-in user", function () {
    const { asFragment } = render(
        <BrowserRouter>
            <Home currUser={user} />
        </BrowserRouter>);
    expect(screen.getByText("Welcome to the Gardener App.")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});