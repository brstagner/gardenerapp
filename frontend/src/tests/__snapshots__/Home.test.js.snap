// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for logged-in user 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h4>
      Welcome to the Gardener App.
    </h4>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for no logged-in user 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="row gy-5"
      >
        <div
          class="col"
        >
          <div
            class="p-3"
          >
            <h4
              class="d-flex justify-content-center"
            >
              Have an account?
            </h4>
            <div
              class="d-flex justify-content-center"
            >
              <a
                class="btn btn-primary d-inline-block"
                href="/login"
              >
                Login
              </a>
            </div>
          </div>
          <div
            class="p-3"
          />
          <div
            class="p-3"
          >
            <h4
              class="d-flex justify-content-center"
            >
              New user?
            </h4>
            <div
              class="d-flex justify-content-center"
            >
              <a
                class="btn btn-primary d-inline-block"
                href="/register"
              >
                Register
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
